src/main.c

```diff
@@ -70,6 +91,7 @@ int main(int argc, char **argv)
 
     loop_cleanup:
         free(user_input);
+        //free(p);
```

Comment:
That crash needs to be fixed. It should not cause a crash if the memory allocations are appropriately sized, and there are no buffer overflows.